### DISCUSSION TITLE
C syntax highlight for "#if 0"

### DISCUSSION THREAD

**Discussion by pandysong** [helix-editor/helix#10436] [General]
### Summary

The code inside "#if 0" is not greyed out.

### Reproduction Steps

"""
#include <maith.h>
#include <stdio.h>

int main()
{

#if 0
    int i;
    int j;
    int k;
    int l;
    int m;
    i = 1;
    j = i +1;
#endif
    int a;
    return 0;
}
"""

Open above code in hx

I expected this to happen (what happened in vim)

![image](https://user-images.githubusercontent.com/5617852/204996695-88f535db-74ff-42ca-aa22-706eecfeabf5.png)

Instead, this happened:

![image](https://user-images.githubusercontent.com/5617852/204996798-7f5a48b6-fcb5-476a-a1c1-db25d2262e95.png)

### Platform

macOS/Linux

### Terminal Emulator

tmux

### Helix Version

helix 22.08.1 (590a6284)

**Comment by pascalkuthe** [comment]
Helix uses treesitter grammars for syntax highlighting which can currently not display semantic information. 
This might theoretically be implemented with stackgraphs #1252 but the problem in practice is that most defines either come from CLI compiler flags or header files (and the include directories are once again defined by compiler flags) that makes this pretty hard to implement.

Other editors achieve this using LSP semantic highlighting (see #814)  but that is not supported in helix (and there is hesitancy about supporting it for performance reasons).
